### ISSUE TITLE
[53] LaTr: Layout-Aware Transformer for Scene-Text VQA

### ISSUE THREAD

**Issue by dhkim0225** [dhkim0225/1day_1paper#81]
Text VQA 쪽에서 성능 엄청 올린 논문 (Fair comparison 을 지향하는 논문은 아님)
+7.6% on TextVQA
+10.8% on ST-VQA
+4.0% on OCR-VQA

[paper](https://arxiv.org/pdf/2112.12494.pdf)

BROS 인용 안해서 마음에 안듦.
거두절미하고 본론으로 ㄱㄱ

# method
Industrial Document Library (IDL) dataset pretrain (https://www.industrydocuments.ucsf.edu/)
13M document, 64M page
데이터는 대충 이런 느낌?
![image](https://user-images.githubusercontent.com/16400591/147734175-44123720-3a56-4195-80e9-cbcd43aeb6a6.png)
아마존 Textract OCR 사용해서 박스정보 전부 뽑아냄.
## Model
모델은 pretrained T5. enc-dec 구조가 전부 잡혀있고, 확장하기가 편했다고 함
![image](https://user-images.githubusercontent.com/16400591/147734419-d38bd8a4-92d1-43db-b15d-1804b1ec6d3d.png)

## Pretraining
pretraining 은 다음 순서로 진행
1. MLM random 하게 15% box 2d span mask 함. ~(아니 이러면서 BROS 인용 안했음)~
2. text token Cross-Entropy 로 학습

masking 할 때,
reading order 에 맞춰서 박스들의 span들을 없애는 방식이고,
없앤 span의 min LT, max RB 구해서 그녀석들의 position 정보를 embedding에 사용함
text token 은 special 로 대체,
box 토큰은 span 전부 감싸는 bbox 로 주는 것.
![image](https://user-images.githubusercontent.com/16400591/147736980-e673a73a-8d32-4e36-acfe-9d8602e56c44.png)

embedding은 그냥 다 합쳐버림
![image](https://user-images.githubusercontent.com/16400591/147735638-b0502681-14b0-4fb3-99a7-9bfb812dfc4f.png)
x0,y0,x1,y2 는 TL/BR 형태 bbox
![image](https://user-images.githubusercontent.com/16400591/147735700-393c8181-baad-4fbd-aa1b-a13298563754.png)

## Finetuning
finetuning 할 때는 
1. pretrained vit 로 visual feature 뽑고
2. question T5 out, (1) out, 학습한 pretrained T5 out 이용해서 enc-dec 학습시킴
자세한 내용은 위 그림 참조

# Result
‡ 모델은 ST-VQA, Text-VQA 데이터 동시 학습한 모델들임
## Text VQA
![image](https://user-images.githubusercontent.com/16400591/147734373-0998cce6-0d2b-4223-b7ca-71a76a0c6233.png)

## ST VQA
![image](https://user-images.githubusercontent.com/16400591/147738104-1ba9eaf7-1cbf-466f-adcb-688f7e5b8aa2.png)

다른 여러가지 실험은 생략
